tests/stages.js

```diff
@@ -0,0 +1,217 @@
+			executor: 'ramping-vus',
+			stages: [
+				{ duration: "0.1s", target: 1 },
+				{ duration: "0.1s", target: 1 }
+			],
+			exec: 'testTagWithSteady'
+		},
+		'tagWithRampDown': {
+			executor: 'ramping-vus',
+			stages: [
+				{ duration: "0.1s", target: 2 },
+				{ duration: "0.1s", target: 1 }
+			],
+			exec: 'testTagWithRampDown'
+		},
+	'unsupported-executor': {
+			executor: 'shared-iterations',
+			iterations: 1,
+			exec: 'testUnsupportedExecutor'
+		}
+	}
+}
+
+export function testGetCurrentStageIndex() {
+	describe('getCurrentStageIndex', () => {
+		describe('returns the first stage when is in t0', () => {
+			expect(getCurrentStageIndex()).to.be.equal(0)
+		})
+
+		sleep(1.2)
+
+		describe('returns the next stage if the iteration exceedes the expected duration', () => {
+			expect(getCurrentStageIndex()).to.be.equal(1)
+		})
+
+		sleep(1)
+
+		describe('returns the last stage when the iteration exceedes the total expected duration', () => {
+			expect(getCurrentStageIndex()).to.be.equal(1)
+		})
+	})
+}
+
+export function testTagWithCurrentStageIndex() {
+	describe('tagWithCurrentStageIndex', () => {
+		describe('tags with the current stage', () => {
+			tagWithCurrentStageIndex()
+			expect(exec.vu.tags['stage']).to.be.equal(`${exec.scenario.iterationInTest}`)
+		})
+
+		// it forces to get only two iterations
+		sleep(1)
+	})
+}
+
+export function testTagWithRampUpProfileWhenOnlyOne() {
+  describe('tagWithCurrentStageProfile', () => {
+		describe(`tags with ramp-up profile when only one stage`, () => {
+			tagWithCurrentStageProfile()
+			expect(exec.vu.tags['stage']).to.be.equal('ramp-up')
+		})
+	})
+}
+
+export function testTagWithRampUp() {
+  describe('tagWithCurrentStageProfile', () => {
+		describe(`tags with ramp-up profile when the current stage has a target greater than previous`, () => {
+			if (exec.vu.iterationInScenario !== 1) {
+				return
+			}
+			tagWithCurrentStageProfile()
+			expect(exec.vu.tags['stage']).to.be.equal('ramp-up')
+		})
+		sleep(0.1)
+	})
+}
+
+export function testTagWithSteady() {
+  describe('tagWithCurrentStageProfile', () => {
+		describe(`tags with steady profile when the current stage has the same target of the previous`, () => {
+			if (exec.vu.iterationInScenario !== 1) {
+				return
+			}
+			tagWithCurrentStageProfile()
+			expect(exec.vu.tags['stage']).to.be.equal('steady')
+		})
+		sleep(0.1)
+	})
+}
+
+export function testTagWithRampDown() {
+describe('tagWithCurrentStageProfile', () => {
+	describe(`tags with ramp-down profile when the current stage has a target less than previous`, () => {
+		if (exec.vu.iterationInScenario !== 1) {
+			return
+		}
+		tagWithCurrentStageProfile()
+		expect(exec.vu.tags['stage']).to.be.equal('ramp-down')
+	})
+	sleep(0.1)
+})
+}
+
+export function testUnsupportedExecutor() {
+	describe('throw Error if the executor is not supported', () => {
+		describe('getCurrentStageIndex', () => {
+			expect(getCurrentStageIndex()).to.throw()
+		})
```

Comment:
Apparently, `throw()` doesn't work as expected :thinking:, I'm going to research more in the meanwhile.